### PR TITLE
Adding dynamic replay destination mode

### PR DESCRIPTION
**Pull request by blinktag**

### CODE REVIEW

File: backends/azure-eventhub/dynamic.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+			if opts.AzureEventHub.PartitionKey != "" {
+				event.PartitionKey = &opts.AzureEventHub.PartitionKey
+			}
+
+			if err := client.Send(ctx, event, sendOpts...); err != nil {
+				llog.Errorf("Unable to replay message: %s", err)
+				break
+			}
+
+			llog.Debugf("Replayed message to Azure Servuce Bus for replay '%s'", outbound.ReplayId)
```
Comment (dselans) [inline backends/azure-eventhub/dynamic.go:51]:
```suggestion
			llog.Debugf("Replayed message to Azure Service Bus for replay '%s'", outbound.ReplayId)
```

File: backends/azure/dynamic.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+				}
+			} else {
+				// Publishing to topic
+				if err := topic.Send(ctx, msg); err != nil {
+					llog.Errorf("Unable to replay message: %s", err)
+					break
+				}
+			}
+
+			llog.Debugf("Replayed message to Azure Servuce Bus for replay '%s'", outbound.ReplayId)
```
Comment (dselans) [inline backends/azure/dynamic.go:79]:
```suggestion
			llog.Debugf("Replayed message to Azure Service Bus for replay '%s'", outbound.ReplayId)
```

**Review by dselans — APPROVED** [review]
Awesome!

File: backends/azure-eventhub/dynamic.go (hunk 1 above)
Comment (blinktag) [reply backends/azure-eventhub/dynamic.go:51]: Fixed!

File: backends/azure/dynamic.go (hunk 2 above)
Comment (blinktag) [reply backends/azure/dynamic.go:79]: Fixed!

**Pull request merged**